components/chatbot/inputBox.js

```diff
@@ -1,41 +1,55 @@
-import React from "react";
-import { useState, useMemo, useEffect, useRef } from "react";
-import { PiArrowDown } from "react-icons/pi";
-import { FaPaperPlane, FaTrashCan, FaRegComments } from "react-icons/fa6";
+import React, { forwardRef, useImperativeHandle } from "react";
+import { FaPaperPlane, FaTrashCan } from "react-icons/fa6";
 import { useRouter } from "next/router";
 import LoadingDots from "../animation/loadingDots";
 import LottieAnimation from "../animation/lottie-animation";
 import documentlottie from "../../public/document-loading.json";
 
+const MAX_INPUT_LENGTH = 6000;
+
+const Instruction = [
+  {
+    title: "Casual Conversation",
+    text: "Tell me a random fun fact about the Roman Empire.",
+  },
+  {
+    title: "Chat with your Documents",
+    text: "From my files, [your custom query]",
+  },
+  {
+    title: "Retrieve Specific Documents",
+    text: "Find me documents related to [Your Specific Detail]",
+  },
+  {
+    title: "On-the-Spot Info Fetch",
+    text: "What is the weather today in Irvine, CA?",
+  },
+];
+
 function InputBoxComponent({
   messageLength,
-  inputText,
-  setInputText,
   isSendChatLoading,
   isGetChatLoading,
+  isClearChatLoading,
   handleClick,
   handleRefresh,
-}) {
+}, ref) {
   const router = useRouter();
-  const messagesEndRef = useRef(null);
-  const Instruction = [
-    {
-      title: "Casual Conversation",
-      text: "Tell me a random fun fact about the Roman Empire.",
-    },
-    {
-      title: "Chat with your Documents",
-      text: "From my files, [your custom query]",
-    },
-    {
-      title: "Retrieve Specific Documents",
-      text: "Find me documents related to [Your Specific Detail]",
+  const [inputText, setInputText] = useState("");
```

Comment:
Keep inputText state within the component, reduce the unnecessary  re-render of its parent.